### PR TITLE
(fix) add Anthropic dependency for Vertex

### LINKED ISSUE #3505: [Bug]: VertexAI integration with Anthropic fails due to missing dependencies
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

When attempting to use the VertexAI integration with Anthropic in OpenDevin, the application fails due to missing Python dependencies.

## Error Message
The following error is encountered:

```python
ModuleNotFoundError: No module named 'anthropic'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 2052, in completion
    model_response = vertex_ai_anthropic.completion(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/litellm/llms/vertex_ai_anthropic.py", line 268, in completion
    raise VertexAIError(
litellm.llms.vertex_ai_anthropic.VertexAIError: vertexai import failed please run `pip install -U google-cloud-aiplatform "anthropic[vertex]"`

### Current OpenHands version

```bash
used with tag :latest: so i suppose 0.8.3
```

### Installation and Configuration

```bash
Running the command i always run with opendevin, this one : 
WORKSPACE_BASE=/mnt/c/Users/tamou/langchain/workspace && docker run --pull=always -e SANDBOX_USER_ID=$(id -u) -e PERSIST_SANDBOX="true" -e SSH_PASSWORD="12345678" -e GOOGLE_APPLICATION_CREDENTIALS=/mnt/c/Users/tamou/langchain/aerial-valor-428210-u3-e27544061b85.json -e VERTEXAI_PROJECT=aerial-valor-428210-u3 -e VERTEXAI_LOCATION=europe-west1 -e LLM_MODEL=vertex_ai/claude-3-5-sonnet@20240620 -e WORKSPACE_MOUNT_PATH=$WORKSPACE_BASE -v $WORKSPACE_BASE:/opt/workspace_base -v /var/run/docker.sock:/var/run/docker.sock -p 3000:3000 --add-host host.docker.internal:host-gateway ghcr.io/opendevin/opendevin:latest
```

### Model and Agent

Default config, didnt get to this part since didnt have any answer

### Operating System

WSL on Windows

### Reproduction Steps

1- Set up the Docker environment with VertexAI and Anthropic configurations.
2- Attempt to run the OpenDevin application.
3- Encounter the above error when the application tries to use the VertexAI integration.

### Logs, Errors, Screenshots, and Additional Context

Any guidance on how to properly set up the VertexAI integration with Anthropic in the Docker environment would be greatly appreciated.

### PR DESCRIPTION
**Pull request by tobitege**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Attempting to use Anthropic models via Vertex fails with dependency error.

Fixes #3505 (ideally 😁 )

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Added an additional dependency to the `pyproject.toml` file.

---
**Other references**

https://docs.anthropic.com/en/api/claude-on-vertex-ai#install-an-sdk-for-accessing-vertex-ai

### CODE REVIEW

File: openhands/runtime/utils/runtime_templates/Dockerfile.j2 [hunk 1]
```diff
@@ -56,6 +56,7 @@ RUN cd /openhands/code && \
     apt-get update && \
     /openhands/miniforge3/bin/mamba run -n base poetry run pip install playwright && \
     /openhands/miniforge3/bin/mamba run -n base poetry run playwright install --with-deps chromium && \
+    /openhands/miniforge3/bin/mamba run -n base poetry run pip install -U google-cloud-aiplatform "anthropic[vertex]" && \
```
Comment (xingyaoww) [inline openhands/runtime/utils/runtime_templates/Dockerfile.j2:59]:
```suggestion
```

I'm a bit hesitant to add these explicitly in the `Dockerfile.j2` (will be hard to maintain in the long run?) - I thought adding things to `pyproject.toml` should be sufficient to resolve it?

Comment (tobitege) [reply openhands/runtime/utils/runtime_templates/Dockerfile.j2:59]: I wasn't quite sure if one was enough. I'll revert the one in the template as per your suggestion and we can see how the pyproject works for the user.

**Review by neubig — COMMENTED** [review]
Actually @tobitege , I wonder if we want to make this and google cloud optional dependencies in a separate group?

It's a minority of users that would want to use these, so we might want to only install them for these users.

**Comment by tobitege** [comment]
> Actually @tobitege , I wonder if we want to make this and google cloud optional dependencies in a separate group?
> 
> It's a minority of users that would want to use these, so we might want to only install them for these users.

Hmm how, though?

**Comment by tobitege** [comment]
@xingyaoww if the user's sandbox config has the `runtime_extra_deps` defined, could that be used to make it into their runtime build?
Similar to some evals using it like:
`        runtime_extra_deps='$OD_INTERPRETER_PATH -m pip install -U google-cloud-aiplatform "anthropic[vertex]"',`

**Comment by xingyaoww** [comment]
@tobitege yes i think so! though i think this is probably over-complicate thing a bit. Also, i think the original issue is not related to dependencies *inside runtime*, but rather dependencies on the backend?

@neubig Even if we add this as an optional group, it is probably a bit tricky for user to install this optional dependency group when they are using the `docker run` (e.g., the docker is already pre-build and hard to inject dependencies into).

I feel this dependency won't add too much size to our package, and it is more like a dependency of litellm for people to use vertex AI?

**Comment by neubig** [comment]
OK, fair enough. I'm generally less worried about size and more worried about dependency conflicts, but both of these packages are both probably slim wrt that as well, so I think it's OK!

**Review by xingyaoww — APPROVED** [review]
Then it LGTM!

**Pull request merged**